File: Swift/factorial.swift
//factorial.swift
// This program takes a number and returns its factorial.

var factorial: Int = 1

print("Please enter only integer")
print("Enter the number you want to get factorial")
let a: String! = readLine()
var numA: Int! = Int(a)

for i in 1...numA {
    factorial *= i
}

print("The factorial of \(a!) is: \(factorial)")

File: Clojure/factorial.clj
(defn factorial []
	(println "Enter a number you want to factorial")
	(let [n (Integer/parseInt (read-line))]
	(loop [output 1 x n]
		(if (= x 0) ; when this is true
			(println output);then return output
			(recur (* output x) (- x 1)) ; otherwise add it
			))
))

(factorial)
